pkg/controller/opentelemetrycollector/reconcile_service.go

```diff
@@ -38,15 +38,35 @@ func (r *ReconcileOpenTelemetryCollector) reconcileService(ctx context.Context)
 
 func service(ctx context.Context) *corev1.Service {
 	instance := ctx.Value(opentelemetry.ContextInstance).(*v1alpha1.OpenTelemetryCollector)
-	name := resourceName(instance.Name)
+	logger := ctx.Value(opentelemetry.ContextLogger).(logr.Logger)
 
-	labels := commonLabels(ctx)
-	labels["app.kubernetes.io/name"] = name
+	name := resourceName(instance.Name)
+	labels := serviceLabels(ctx, name)
 
 	// by coincidence, the selector is the same as the label, but note that the selector points to the deployment
 	// whereas 'labels' refers to the service
 	selector := labels
 
+	ports := instance.Spec.Ports
```

Comment:
Might be better if user only has to provide additional ports - so by default it will infer as many as possible?

Reply:
I had a couple of thoughts when deciding to override, but I think they don't make much sense and you might be right. What I had in mind was:

1. what if the user wants to suppress exposing a port? But then, why would a user then use the receiver?
1. what if we are inferring the wrong port for a receiver? We would end up with a port that is being unused, which is kind of against one of the security commandments ("open only the ports you need")
1. auto-pilot should stay off when the user has decided to take command -- following this idea, we should stop trying to be smart, and let the user have full control over the exposed ports

I'm open to change the PR, so that the ports specified are in addition to the inferred.

Reply:
I think it would be more user friendly if they didn't need to know what ports to add, unless they really needed to  - i.e. when adding a new receiver that is not currently known by the operator.

However we should cater for overrides, in the rare situation a receiver changes its default port and the operator hasn't yet been updated.

Reply:
For now, let's just change the list to be additive, and see if there's a need in the community to have a flag, indicating that the list of ports is definitive.